integrated_channels/sap_success_factors/utils.py

```diff
@@ -5,32 +5,164 @@
 
 import datetime
 import json
+import os
+from logging import getLogger
 
 from django.apps import apps
 from django.utils import timezone
+from six.moves.urllib.parse import urlunparse  # pylint: disable=import-error,wrong-import-order
 
+from enterprise.django_compatibility import reverse
 from enterprise.lms_api import parse_lms_api_datetime
 from integrated_channels.integrated_channel.course_metadata import BaseCourseExporter
 
 
+LOGGER = getLogger(__name__)
+COURSE_URL_SCHEME = os.environ.get('SUCCESSFACTORS_COURSE_EXPORT_DEFAULT_URL_SCHEME', 'https')
 UNIX_EPOCH = datetime.datetime(1970, 1, 1, tzinfo=timezone.utc)
 UNIX_MIN_DATE_STRING = '1970-01-01T00:00:00Z'
 UNIX_MAX_DATE_STRING = '2038-01-19T03:14:07Z'
+SUCCESSFACTORS_OCN_LANGUAGE_CODES = {
+    "ms": {
+        "_": "Malay"
+    },
+    "ar": {
+        "_": "Arabic"
+    },
+    "bg": {
+        "_": "Bulgarian"
+    },
+    "cs": {
+        "_": "Czech"
+    },
+    "cy": {
+        "_": "Welsh"
+    },
+    "da": {
+        "_": "Danish"
+    },
+    "de": {
+        "_": "German"
+    },
+    "el": {
+        "_": "Greek"
+    },
+    "en": {
+        "ca": "English Canadian",
+        "gb": "English United Kingdom",
+        "_": "English"
+    },
+    "es": {
+        "mx": "Spanish Mexican",
+        "_": "Spanish"
+    },
+    "fi": {
+        "_": "Finnish"
+    },
+    "fr": {
+        "ca": "French Canadian",
+        "_": "French"
+    },
+    "hi": {
+        "_": "Hindi"
+    },
+    "hr": {
+        "_": "Croatian"
+    },
+    "hu": {
+        "_": "Hungarian"
+    },
+    "in": {
+        "_": "Indonesian"
+    },
+    "it": {
+        "_": "Italian"
+    },
+    "iw": {
+        "_": "Hebrew"
+    },
+    "ja": {
+        "_": "Japanese"
+    },
+    "ko": {
+        "_": "Korean"
+    },
+    "nl": {
+        "_": "Dutch"
+    },
+    "no": {
+        "_": "Norwegian"
+    },
+    "pl": {
+        "_": "Polish"
+    },
+    "pt": {
+        "br": "Brazilian Portuguese",
+        "_": "Portuguese"
+    },
+    "ro": {
+        "_": "Romanian"
+    },
+    "ru": {
+        "_": "Russian"
+    },
+    "sk": {
+        "_": "Slovak"
+    },
+    "sl": {
+        "_": "Slovenian"
+    },
+    "sr": {
+        "_": "Serbian"
+    },
+    "sv": {
+        "_": "Swedish"
+    },
+    "th": {
+        "_": "Thai"
+    },
+    "tr": {
+        "_": "Turkish"
+    },
+    "uk": {
+        "_": "Ukrainian"
+    },
+    "vi": {
+        "_": "Vietnamese"
+    },
+    "zh": {
+        "hk": "Chinese Hong Kong",
+        "tw": "Chinese Taiwan",
+        "_": "Chinese"
+    }
+}
 
 
-class SapCourseExporter(BaseCourseExporter):
+class SapCourseExporter(BaseCourseExporter):  # pylint: disable=abstract-method
     """
     Class to provide data transforms for SAP SuccessFactors course export task.
     """
 
     def __init__(self, user, plugin_configuration):
         super(SapCourseExporter, self).__init__(user, plugin_configuration)
 
-    def get_serialized_data(self):
```

Comment:
I don't think we want to here. `get_serialized_data` is still the canonical form for integrated channels in general; we just have to do `get_serialized_data_blocks` here because we need to page through a given number of elements at a time.